### DISCUSSION TITLE
How to use oh-my-posh with Constrainedlanguage mode?

### DISCUSSION THREAD

**Discussion by alexanderwwagner** [JanDeDobbeleer/oh-my-posh#3914] [Troubleshoot] [answered]
Hello together,

**Description**
for me it's not possible to install oh-my-posh with one of the described methods... because of the
ConstrainedLanguage mode. I did some research but I could not found a working solution.

**Setup**
- Windows 10
- Powershell 7
- ConstrainedLanguage

**Reproduce**
Try to install with the manual method:
```
PS C:\Users\wagneral> Set-ExecutionPolicy Bypass -Scope Process -Force; Invoke-Expression ((New-Object System.Net.WebClient).DownloadString('https://ohmyposh.dev/install.ps1'))
New-Object: Cannot create type. Only core types are supported in this language mode.
```

I can install oh-my-posh with this command if I start a powershell with local admin... but I'm not able to install it for my normal user.

Do you have a idea what I can do to get oh-my-posh working?
Thank you!

**Comment by JanDeDobbeleer** [comment]
Why not install from the Windows Store in this case?

**Reply by alexanderwwagner** [reply]
The store is disabled by my admins... 
(GUI and cli)

**Reply by JanDeDobbeleer** [reply]
Can you download and run the installer from the releases tab? Otherwise it's a complete manual install by downloading the binary (unsigned) and managing the PATH and everything yourself.

**Reply by alexanderwwagner** [reply]
Unfortunately that doesn't work either because I can only start installer as admin...
In the meantime I tried to copy the oh-my-posh folder which includes the nessessary binarys to my local user and add the directory to my path variable.
After that I insert the following phrase in my profile file.
`oh-my-posh init pwsh --config "$env:POSH_THEMES_PATH\jandedobbeleer.omp.json"`

If I restart my powershell I get following error:
`Microsoft.PowerShell_profile.ps1: Cannot dot-source this command because it was defined in a different language mode. To invoke this command without importing its contents, omit the '.' operator.`

Is it possible to solve this?

**Reply by JanDeDobbeleer** [reply]
@alexanderwwagner you probably need to change jobs at this point in time. We source a script, something that works in constrained language mode, but probably not for all locked down cases. That's not something we can solve on our end. You could try to save the init script (oh-my-posh init pwsh --print) elsewhere and load that.

**Reply by alexanderwwagner** [reply]
Hmm if I call a "HelloWorld" powershell script within my $PROFILE configuration it works out of the box.
Microsoft.PowerShell_profile.ps1
`./helloworld.ps1`

Console output:
```
PowerShell 7.3.4
Hello, World!
Hello, World!
PS C:\Users\wagneral>
```

If I copy the inserts of "oh-my-posh" init pwsh --print") to a new powershellscript and start this within my $PROFILE configuration
I have again:
`Microsoft.PowerShell_profile.ps1: Cannot dot-source this command because it was defined in a different language mode. To invoke this command without importing its contents, omit the '.' operator.`

Have you any idea on which point in your script I get the problem with the language mode?

**Reply by JanDeDobbeleer** [reply]
@alexanderwwagner no, that's a PowerShell question that goes beyond my current knowledge.

**Reply by alexanderwwagner** [reply]
Hmm now I am one step further...
If I change my "Microsoft.PowerShell_profile.ps1" to use a hardcoded path (not a env) I can run oh-my-posh!
`oh-my-posh init pwsh --config 'C:\Users\wagneral\AppData\Local\Packages\oh-my-posh\themes\jandedobbeleer.omp.json' | Invoke-Expression`

After that I manually install the recommenend font "Meslo LGM NF" in a manually way for all users.
Then I changed the font in "Windows Terminal" to use this:
![image](https://github.com/JanDeDobbeleer/oh-my-posh/assets/45471644/1bbbcd7f-b6c0-4cb4-baab-52350a359577)

But for any reason my symbols are not working... do I have to set manually some path ENVs?
![image](https://github.com/JanDeDobbeleer/oh-my-posh/assets/45471644/f79bc3bc-dc8a-4254-a4d4-222b398d7242)

Thank you again!

**Reply by JanDeDobbeleer** [reply]
@alexanderwwagner probably the font name is incorrect. Or the WT configuration.

**Comment by alexanderwwagner** [comment, marked as answer]
So finally I get it working!

1. Installed it with the manual method of the documentation (oh-my-posh) as admin
2. Copied installation directories inclusive all binarys to a path to which my normal user has access
3. Add path of binary to "ENV PATH variable"
4. Write in the absolute path without env variable in my $PROFILE
`oh-my-posh init pwsh --config 'C:\Users\wagneral\AppData\Local\Packages\oh-my-posh\themes\jandedobbeleer.omp.json' | Invoke-Expression`

5. Install recommended font like specified in the documentation of oh-my-posh

6. Install WindowsTerminal (manually from github) 
7. Set Font to installed font of step 5
![grafik](https://github.com/JanDeDobbeleer/oh-my-posh/assets/45471644/dd3dc7d3-bf22-461f-913e-64337e4ad1fd)

8. Set locale to UTF8
![grafik](https://github.com/JanDeDobbeleer/oh-my-posh/assets/45471644/d06427a8-2e13-41dd-8140-b3057d68856a)

9. Add registry insert to can open Windows terminal from the context menu like Powershell 7 (Normal/Admin)

@JanDeDobbeleer thank you for your support! :)

**Reply by thwyster** [reply]
thanks for this aswer, you resolve my problem too!

**Comment by cspargo-healthedge** [comment]
For me I just had to set the Locale to UTF-8 and it started working again 🥂
